flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/CatalogLoader.java

```diff
@@ -21,10 +21,12 @@
 
 import java.io.Serializable;
 import java.util.Map;
+import org.apache.flink.table.factories.CatalogFactory;
 import org.apache.hadoop.conf.Configuration;
```

Comment:
Looks like we need to remove the import.

Reply:
We can eventually if we want to strip out references from the existing Hadoop and Hive catalog implementations.  As it stands, the `Configuration` class is only loaded if a catalog ends up calling `HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration())` which is avoided for catalogs which are loaded by passing `catalog-impl` AND when `org.apache.hadoop.conf.Configuration` isn't available to Flink's application level class loader (even it it is available to the user lib classloader).

Reply:
HI @kainoa21 ,

we create  catlog like this 

   tableEnv.executeSql("CREATE CATALOG iceberg WITH (\n" +
                "  'type'='iceberg',\n" +
                "  'warehouse'='s3://ka-app-code/icebergst/',\n" +
                "  'catalog-impl'='org.apache.iceberg.aws.glue.GlueCatalog',\n" +
                "  'io-impl'='org.apache.iceberg.aws.s3.S3FileIO'\n" +
                ")");

we are using flink 1.13.2 and 

<dependency>
			<groupId>org.apache.iceberg</groupId>
			<artifactId>iceberg-flink-runtime-1.13</artifactId>
			<version>0.13.1</version>
		</dependency>

		<!-- https://mvnrepository.com/artifact/software.amazon.awssdk/bundle -->
		<dependency>
			<groupId>software.amazon.awssdk</groupId>
			<artifactId>bundle</artifactId>
			<version>2.15.40</version>
		</dependency>

		<!-- https://mvnrepository.com/artifact/software.amazon.awssdk/url-connection-client -->
		<dependency>
			<groupId>software.amazon.awssdk</groupId>
			<artifactId>url-connection-client</artifactId>
			<version>2.15.40</version>
		</dependency>


but we still get this error 


```
java.lang.NoClassDefFoundError: org/apache/hadoop/conf/Configuration
	at org.apache.iceberg.flink.FlinkCatalogFactory.clusterHadoopConf(FlinkCatalogFactory.java:160)
	at org.apache.iceberg.flink.FlinkCatalogFactory.createCatalog(FlinkCatalogFactory.java:126)
	at org.apache.flink.table.factories.FactoryUtil.createCatalog(FactoryUtil.java:267)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.createCatalog(TableEnvironmentImpl.java:1292)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeInternal(TableEnvironmentImpl.java:1122)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeSql(TableEnvironmentImpl.java:742)
	at basic.application.StreamingJob.main(StreamingJob.java:196)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:355)
	at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:222)
	at org.apache.flink.client.ClientUtils.executeProgram(ClientUtils.java:114)
	at org.apache.flink.client.cli.CliFrontend.executeProgram(CliFrontend.java:812)
	at org.apache.flink.client.cli.CliFrontend.run(CliFrontend.java:246)
	at org.apache.flink.client.cli.CliFrontend.parseAndRun(CliFrontend.java:1054)
	at org.apache.flink.client.cli.CliFrontend.lambda$main$10(CliFrontend.java:1132)
	at org.apache.flink.runtime.security.contexts.NoOpSecurityContext.runSecured(NoOpSecurityContext.java:28)
	at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:1132)
```

this method is still called 

 ```
public static Configuration clusterHadoopConf() {
    return HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration());
  }
```

please advice